src/hera/workflows/runner.py

```diff
@@ -1,6 +1,31 @@
 """The entrypoint for the Hera Runner when running on Argo."""
 
+import os
+from typing import Any, Optional, Tuple, Union
+
 from hera.workflows._runner.util import _run
 
+
+class RunnerException(Exception):
+    """An exception class that can hold outputs to write to file on Argo before passing the exception up.
+
+    Note: Must be enabled via hera.shared.global_config.experimental_features.
+    """
+
+    def __init__(self, outputs: Union[Tuple[Any], Any], exit_code: Optional[int] = None) -> None:
```

Comment:
We should keep exit code as the first param followed by details.

Reply:
What if I want the original exception (and exit code) to bubble up? i.e. in `src/hera/workflows/_runner/util.py`:

```py
            except RunnerException as e:
                _save_annotated_return_outputs(e.outputs, output_annotations)
                if e.exit_code is not None:
                    exit(e.exit_code)

                raise e
```

I wouldn't really say `exit_code` is necessary in `RunnerException` as then you should use `RunnerOutput`. If we remove it then we can avoid "two ways to do one thing".